docs/xml-sitemaps.md

```diff
@@ -8,160 +8,51 @@ The sitemap index can hold a maximum of 50000 sitemaps, and a single sitemap can
+The XML Sitemaps in Altis are provided by our integration with [Yoast SEO](https://developer.yoast.com/features/xml-sitemaps/overview). A sitemap index is automatically linked to in the site's [robots.txt](./robots-txt.md) file at `/sitemap_index.xml` so you don't need to manually submit them to the [Google Search Console](https://search.google.com/search-console/). You can resubmit them from the search console and get diagnostic information there at any time.
 
 **Note:** Sitemaps are only linked to in the `robots.txt` file if the site is public and in production. To debug locally you will need to define the constant `HM_DISABLE_INDEXING` as `false`.
 
 ## Google Site Verification
 
 It is necessary to [verify the site with Google Search Console](https://support.google.com/webmasters/answer/9008080?hl=en) before you can access information about your site's search results performance. It is recommended to use the HTML file upload solution by committing the file to your project's root directory, although it is possible to add the meta tag by filling in the [verification code on the Reading Settings page in the admin](admin://options-reading.php).
 
-## Adding Custom Sitemaps
-Sitemaps are provided for built-in content types like pages and author archives out of the box. If you want to add a custom sitemap with additional features, such as a video sitemap, you can register a custom sitemap provider.
+## Adding or Removing Post Types
+Sitemaps are provided for public content types like posts and pages out of the box. If you want to add support for additional custom post types in your sitemaps, you can do so by using the built-in Yoast SEO filters.
 
-To do so, create a custom PHP class that extends the abstract `WP_Sitemaps_Provider` class. Then you can use the `wp_register_sitemap_provider()` function to register it.
-
-The example below shows a minimal implementation for a custom video sitemap:
+To add a post type to the sitemaps, use the `register_sitemap` method of the `$wpseo_sitemaps` global like in the example below:
 
 ```php
-class Video_Sitemap_Provider extends WP_Sitemaps_Provider {
-
-	/**
-	 * Set the name and object type properties. Required.
-	 */
-	public function __construct() {
-		$this->name = 'videos';
-		$this->object_type = 'post';
-	}
+add_action( 'init', function () {
+	global $wpseo_sitemaps;
 
-	/**
-	 * Return the list of URLs for the current page.
-	 */
-	public function get_url_list( $page ) {
-		$videos = new WP_Query( [
-			'post_type' => 'video',
-			// phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_numberposts
-			'posts_per_page' => 2000,
-			'fields' => 'ids',
-			'paged' => $page,
-		] );
-
-		$urls = [];
-
-		foreach ( $videos->posts as $video_id ) {
-			$urls[] = get_post_meta( $video_id, 'video_url', true );
-		}
-
-		return $urls;
+	if ( isset( $wpseo_sitemaps ) && ! empty( $wpseo_sitemaps ) ) {
+		$wpseo_sitemaps->register_sitemap( 'video', 'create_video_sitemap' );
```

Comment:
I had to dig into the actual sitemap code and it's kinda gnarly why this works. Also the official docs aren't great unfortunately. I think we can do a much better job but maybe not right now as I'm not sure how often anyone really wants to make a custom site map.

We should try and make it clear that this is the way to add a post type sitemap for any non public or previously excluded post type, the second parameter can actually be any string, it could be `_` so long as the first parameter is valid post type.

If the first parameter is not a post type, taxonomy or the string 'author' then Yoast SEO will look for any registered `WPSEO_Sitemap_Provider` subclass or use 2nd parameter as a callback function in which case the callback needs to generate the entire sitemap XML. This example code only works because there's a post type provider class that will match the "video" post type.